ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -136,8 +141,16 @@ public AccountStatsMySqlStore(List<MySqlUtils.DbEndpoint> dbEndpoints, String lo
   public void storeStats(StatsWrapper statsWrapper) {
     StatsSnapshot prevSnapshot =
         previousStats == null ? new StatsSnapshot((long) -1, new HashMap<>()) : previousStats.getSnapshot();
-    applyFunctionToContainerUsageInDifferentStatsSnapshots(statsWrapper.getSnapshot(), prevSnapshot,
-        this::tryUpdateStorageUsage);
+    ContainerUsageFunction function = this::tryUpdateStorageUsage;
+    ContainerStorageFunctionBatchUpdater batchUpdater = null;
+    if (config.updateBatchSize >= 0) {
```

Comment:
I would add class variable boolean batchEnabled, set
`this.batchEnabled = config.updateBatchSize >= 0`
here and reference that throughout.